test_scripts/Polices/External_UCS_Informing_HMI/019_ATF_N_GetListofPermissions_empty_due_to_invalid_param_entityType_OnAppPermissionChange.lua

```diff
@@ -0,0 +1,142 @@
+  Test["Precondition_PTU_and_OnAppPermissionConsent_Invalid_"..params_invalid_data[i]] = function(self)
+    local ptu_file_path = "files/jsons/Policies/Related_HMI_API/"
+    local ptu_file = "OnAppPermissionConsent_ptu.json"
+    
+    testCasesForPolicyTable:flow_SUCCEESS_EXTERNAL_PROPRIETARY(self, nil, nil, nil, ptu_file_path, nil, ptu_file)
+    
+    EXPECT_HMINOTIFICATION("SDL.OnAppPermissionChanged",{ appID = self.applications[config.application1.registerAppInterfaceParams.appName]})
+    :Do(function(_,data)
+      if (data.params.appPermissionsConsentNeeded== true) then
+        local RequestIdListOfPermissions = self.hmiConnection:SendRequest("SDL.GetListOfPermissions", {appID = self.applications[config.application1.registerAppInterfaceParams.appName]})
+          EXPECT_HMIRESPONSE(RequestIdListOfPermissions,{result = {code = 0, method = "SDL.GetListOfPermissions",
+            -- allowed: If ommited - no information about User Consent is yet found for app.
+            allowedFunctions = {
+              { name = "Location", id = 156072572},
+              { name = "Notifications", id = 1809526495}
+            },
+            externalConsentStatus = {}
+          }
+        })
+        :Do(function()
+          local ReqIDGetUserFriendlyMessage = self.hmiConnection:SendRequest("SDL.GetUserFriendlyMessage",
+          {language = "EN-US", messageCodes = {"AppPermissions"}})
+          
+          EXPECT_HMIRESPONSE(ReqIDGetUserFriendlyMessage,
+          {result = {code = 0, messages = {{messageCode = "AppPermissions"}}, method = "SDL.GetUserFriendlyMessage"}})
+          :Do(function(_,_)
+            self.hmiConnection:SendNotification("SDL.OnAppPermissionConsent",
+            {
+              appID = self.applications[config.application1.registerAppInterfaceParams.appName],
+              consentedFunctions = {
+                { allowed = true, id = 156072572, name = "Location-1"},
+                { allowed = true, id = 1809526495, name = "Notifications"}
+              },
+              externalConsentStatus = {
+                {entityType = params_invalid_data[i], entityID = 113, status = "ON"}
+              },
+              source = "GUI"
+            })
+            EXPECT_NOTIFICATION("OnPermissionsChange"):Times(0)
+            commonTestCases:DelayedExp(10000)
+          end)
+        end)
+      else
+        commonFunctions:userPrint(31, "Wrong SDL bahavior: there are app permissions for consent, isPermissionsConsentNeeded should be true")
+        return false
+      end
+    end)
+  end
+end
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Test")
+
+function Test:TestStep_GetListofPermissions_entityType_invalid()
+  local RequestIdListOfPermissions = self.hmiConnection:SendRequest("SDL.GetListOfPermissions", {appID = self.applications[config.application1.registerAppInterfaceParams.appName]})
+  
+  EXPECT_HMIRESPONSE(RequestIdListOfPermissions, {
+    code = "0",
+    allowedFunctions = {
```

Comment:
@vvvakulenko, based on comment on line 124, you should use:

allowedFunctions = allowed_func